### PR TITLE
[classlib] Function:plot fix sample alignment with grid lines

### PR DESCRIPTION
**Pull request by mtmccrea**
## Purpose and Motivation
When plotting functions with a short duration, individual samples are visibly misaligned with the grid. The x-axis position of each sample doesn't account for the duration of the last sample period (a "fencepost problem"). This PR sets the sample position of the plotted points explicitly in the case that this time offset would be visible (i.e. a maximally wide plot would contain samples separated by ~2.5 pixels).

### Example
Below, both `SinOsc` and `Impulse` are plotted for a duration of 2 times their cycle period. So we expect the cycles to start at the leftmost part the plot (first cycle) and the center of the plot (second cycle). 

Currently:
Screen Shot 2022-08-28 at 00 44 20Screen Shot 2022-08-28 at 00 45 28
After the fix:
Screen Shot 2022-08-28 at 00 46 58Screen Shot 2022-08-28 at 00 47 24

Note the differences: 1. the second cycle begins at the center of the plot, 2. the _duration_ of the last sample is properly accounted for on the  time grid. The second point is further clarified when using the `plotMode = \steps` to visualize the sample-and-hold nature of discrete-time sampling:
Screen Shot 2022-08-28 at 00 47 06

Reproduce with ([more examples](https://gist.github.com/mtmccrea/dedb79a48312cd197b042d5c884ab44d)):
```
// SinOsc: the second cycle should begin exactly at
// time t = period (centered in the plot). fs = 48kHz.
(
var period = 0.0005, dur = period * 2;
f = { SinOsc.ar(period.reciprocal) };
f.plot(dur, bounds: Window.screenBounds.insetBy(400, 200)).plotMode_(\plines);
)
// Impulse - Control rate
( 
s.waitForBoot{
	var persamps = 6;
	var period = persamps * s.options.blockSize / s.sampleRate; // n samples per cycle
	var nper = 2;
	var dur = period * nper;
	postf("Expecting an impulse at:\n\ttime %\n\tsample %\n", period, nper.collect(_*persamps+1));

	f = { Impulse.kr(period.reciprocal) };
	f.loadToFloatArray(dur, action: {|arr| a = arr });
	f.plot(dur, bounds: Window.screenBounds.insetBy(400, 200)).plotMode_(\plines);
}
)
```

This PR also sneaks in a slight padding of x-grid tick labels, which was needed in some cases when the bounding rect of a `String` underestimates the width of the string (second commit).

## Types of changes

- Bug fix.

## To-do list

- [x] Code is tested
- [x] All tests are passing (`TestPlotter`)
~~[ ] Updated documentation~~
- [x] This PR is ready for review

### CODE REVIEW

File: SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc [hunk 1]
```diff
@@ -1128,31 +1128,52 @@ Plotter {
...
 		action = { |array, buf|
 			var numChan = buf.numChannels;
-			{
+			var numFrames = buf.numFrames;
+			var frameDur;
+
+			defer {
 				plotter.setValue(
-					array.unlace(numChan).collect(_.drop(-1)),
+					array.unlace(numChan),
```
Comment (mtmccrea) [inline SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc:1149]: Note: previously, the last data frame was dropped from in the plot. It's not apparent to me why, and seems to work fine plotting _all_ data frames.

File: SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc [hunk 2]
```diff
@@ -1128,31 +1128,52 @@ Plotter {
...
 				);
+
 				plotter.domainSpecs = ControlSpec(0, duration, units: "Time (s)");
-			}.defer
+
+				// If individual values might be separated by at least 2.5 px
+				// (based on a plot at full screen width), set the x values (domain)
+				// explicitly for accurate time alignment with grid lines.
+				if(numFrames < (Window.screenBounds.width / 2.5)) {
+					frameDur = (this.value.rate == \control).if({
```
Comment (telephon) [inline SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc:1163]: `frameDur = if(this.value.rate == \control) {`  would keep one style
Also no need for semicolons for the return of the functions (e.g. for `1 / server.sampleRate;`)

File: SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc (hunk 1 above)
Comment (telephon) [reply SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc:1149]: yes, this seems much better.

File: SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc (hunk 2 above)
Comment (telephon) [reply SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc:1163]: (but this is just a suggestion)

Comment (mtmccrea) [reply SCClassLibrary/Common/GUI/PlusGUI/Math/PlotView.sc]: Thanks for suggesting this. I thought at the time that setting a variable with receiver syntax read more naturally, but I agree consistency is better.

After reviewing the [discussion](https://github.com/supercollider/supercollider/issues/2914) that led to return statement [guidelines](https://github.com/supercollider/supercollider/wiki/Code-style-guidelines#return-statements), I removed `;` from both return statements in the `if`, and added one after `defer` (that statement doesn't imply return and could be followed by more code).

**Review by telephon — APPROVED** [review]

**Pull request merged**